content/StsServerIdentity/Services/Certificate/KeyVaultCertificateService.cs

```diff
@@ -1,43 +1,79 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Security.Cryptography.X509Certificates;
 using System.Threading.Tasks;
 using Microsoft.Azure.KeyVault;
-using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Microsoft.Azure.KeyVault.Models;
+using Microsoft.Azure.Services.AppAuthentication;
 
 namespace StsServerIdentity.Services.Certificate
 {
-    public class KeyVaultCertificateService : ICertificateService
+    public class KeyVaultCertificateService
     {
-        private readonly string _vaultAddress;
-        private readonly string _vaultClientId;
-        private readonly string _vaultClientSecret;
+        private readonly string _keyVaultEndpoint;
+        private readonly string _certificateName;
 
-        public KeyVaultCertificateService(string vaultAddress, string vaultClientId, string vaultClientSecret)
+        public KeyVaultCertificateService(string keyVaultEndpoint, string certificateName)
         {
-            _vaultAddress = vaultAddress;
-            _vaultClientId = vaultClientId;
-            _vaultClientSecret = vaultClientSecret;
+            if (string.IsNullOrEmpty(keyVaultEndpoint))
+            {
+                throw new ArgumentException("missing keyVaultEndpoint");
+            }
+            
+            _keyVaultEndpoint = keyVaultEndpoint; // "https://damienbod.vault.azure.net"
+            _certificateName = certificateName; // certificateName
         }
 
-        public X509Certificate2 GetCertificateFromKeyVault(string vaultCertificateName)
+        public (X509Certificate2, X509Certificate2) GetCertificatesFromKeyVault()
         {
-            var keyVaultClient = new KeyVaultClient(AuthenticationCallback);
+            (X509Certificate2, X509Certificate2) certs = (null, null);
+            Task<(X509Certificate2, X509Certificate2)> task = Task.Run(async () => await GetCertsAsync());
+            certs = task.Result;
```

Comment:
Why wrap this in `Task.Run`? could you not go `async` all the way? or at least use `GetAwater().GetResult()` instead of `.Result`

Reply:
It is used in the ConfigureServices which is sync, so I have to use the async in sync as far as I understand. Is GetAwater().GetResult() better, my understand is weak here ...

Reply:
now 
keyVaultCertificateService.GetCertificatesFromKeyVault().GetAwaiter().GetResult();